akka-projection-core/src/main/scala/akka/projection/scaladsl/StatefulHandler.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.scaladsl
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Failure
+import scala.util.Success
+
+import akka.Done
+import akka.annotation.ApiMayChange
+
+@ApiMayChange abstract class StatefulHandler[State, Envelope](implicit ec: ExecutionContext) extends Handler[Envelope] {
+
+  private var state: Future[State] = _
+
+  /**
+   * Invoked to load the initial state when the projection is started or if previous `process` failed.
+   */
+  def initialState(): Future[State]
```

Comment:
I was thinking that `initialState` could be parameterless so Scala users at least could implement it as a constant.  I guess that wouldn't work for Java users though.

Reply:
is't typically loading state from db, so I think it makes sense to have `()`. It's still possible to define it as a `val`
```
      override val initialState: Future[Map[Word, Count]] =
        Future.successful(Map.empty)
```